src/jit/gentree.h

```diff
@@ -1122,9 +1122,15 @@ struct GenTree
             return false;
         }
 
-        if (gtOper == GT_NOP || gtOper == GT_CALL)
+        if (gtType == TYP_VOID)
         {
-            return gtType != TYP_VOID;
+// These are the only operators which can produce either VOID or non-VOID results.
+#ifdef FEATURE_SIMD
+            assert(OperIs(GT_NOP, GT_CALL, GT_LOCKADD, GT_FIELD_LIST, GT_COMMA, GT_SIMD));
```

Comment:
Shouldn't relops also be on this list? It seems that a relop node used by a JTRUE is similar to SIMD equality node.

Reply:
Interesting, it looks like the main reason we still need `GTK_NOVALUE` is that various "store" operations use `gtType` to indicate the size of the location to store to and not the type of the value returned by the node, that would be `TYP_VOID`.